wpilibj/src/main/java/edu/wpi/first/wpilibj/controller/PIDController.java

```diff
@@ -0,0 +1,639 @@
+    try {
+      return m_Kp;
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Get the Integral coefficient.
+   *
+   * @return integral coefficient
+   */
+  public double getI() {
+    m_thisMutex.lock();
+    try {
+      return m_Ki;
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Get the Differential coefficient.
+   *
+   * @return differential coefficient
+   */
+  public double getD() {
+    m_thisMutex.lock();
+    try {
+      return m_Kd;
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Return the current controller output.
+   *
+   * <p>This is always centered around zero and constrained to the min and max outputs.
+   *
+   * @return The latest calculated output.
+   */
+  public double getOutput() {
+    m_thisMutex.lock();
+    try {
+      return m_output;
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Set the reference for the PIDController.
+   *
+   * @param reference the desired reference
+   */
+  public void setReference(double reference) {
+    m_thisMutex.lock();
+    try {
+      if (m_maximumInput > m_minimumInput) {
+        m_reference = clamp(reference, m_minimumInput, m_maximumInput);
+      } else {
+        m_reference = reference;
+      }
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Returns the current reference of the PIDController.
+   *
+   * @return the current reference
+   */
+  public double getReference() {
+    m_thisMutex.lock();
+    try {
+      return m_reference;
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Return true if the error is within the percentage of the total input range,
+   * determined by SetTolerance. This asssumes that the maximum and minimum
+   * input were set using SetInput.
+   *
+   * <p>Currently this just reports on target as the actual value passes through the setpoint.
+   * Ideally it should be based on being within the tolerance for some period of time.
+   *
+   * <p>This will return false until at least one input value has been computed.
+   */
+  public boolean atReference() {
+    double error = getError();
+
+    m_thisMutex.lock();
+    try {
+      double deltaError = (error - m_prevError) / getPeriod();
+      if (m_toleranceType == Tolerance.kPercent) {
+        return Math.abs(error) < m_tolerance / 100 * m_inputRange
+               && Math.abs(deltaError) < m_deltaTolerance / 100 * m_inputRange;
+      } else {
+        return Math.abs(error) < m_tolerance
+               && Math.abs(deltaError) < m_deltaTolerance;
+      }
+    } finally {
+      m_thisMutex.unlock();
+    }
+  }
+
+  /**
+   * Set the PID controller to consider the input to be continuous,
+   *
+   * <p>Rather then using the max and min input range as constraints, it considers them to be the
+   * same point and automatically calculates the shortest route to the setpoint.
+   */
+  public void setContinuous() {
+    m_thisMutex.lock();
```

Comment:
Hmm. Wouldn't it be clearer for this to simply delegate to `setContinuous(boolean)`?